### ISSUE TITLE
[CSGO] OnClientCommandKeyValues KeyValues reporting incorrect keys

### ISSUE THREAD

**Issue by Sikarii** [alliedmodders/sourcemod#1790]
# Help us help you
  - [x] I have checked that my issue [doesn't exist yet](https://github.com/alliedmodders/sourcemod/issues).
  - [x] I have tried my absolute best to reduce the problem-space and have provided the absolute smallest test-case possible.
  - [x] I can always reproduce the issue with the provided description below.

# Environment
  * Operating System version: Windows 10 19044.1766, also confirmed on Ubuntu 20.04 by @Szwagi
  * Game/AppID (with version if applicable): 730/740 1.38.3.5, protocol version 13835 [1483/1483]
  * Current SourceMod version: 1.10.0.6544
  * Current SourceMod snapshot: 6544
  * Current Metamod: Source snapshot: 1.11.0-dev+1147 (1147)
  - [x] I have updated SourceMod to the [latest version](https://www.sourcemod.net/downloads.php) and it still happens.
  - [x] I have updated SourceMod to the [latest snapshot](https://www.sourcemod.net/downloads.php?branch=dev) and it still happens.
  - [x] I have updated SourceMM to the [latest snapshot](https://sourcemm.net/downloads.php?branch=dev) and it still happens.

# Description
The KeyValues structure returned in `OnClientCommandKeyValues` seems to return partially incorrect data, the values themselves seem OK, but keys (and section names) seem to be incorrect.

Everything seems to return "GameInfo" as the section name, and arguments being either "title" or "" (an empty string).
Is this a sign of a deeper issue?

# Problematic Code (or Steps to Reproduce)
```c
public Action OnClientCommandKeyValues(int client, KeyValues kv)
{
    char cmd[16];
    kv.GetSectionName(cmd, sizeof(cmd));

    char payload[8192];
    kv.ExportToString(payload, sizeof(payload));

    PrintToServer("Event: %s", cmd);
    PrintToServer("Payload: %s", payload);

    return Plugin_Continue;
}
```

# Logs
<details>
  <summary>Old CrosshairCode</summary>

```
Event: GameInfo
Payload: "GameInfo"
{
        "title"         "CSGO-xxxx-xxxxx-xxxxx-xxxxx-xxxxx"
}
```
</details>

<details>
  <summary>Old ClanTagChanged</summary>

```
Event: GameInfo
Payload: "GameInfo"
{
        "title"         "SM"
        ""              "SourceMod"
}
```
</details>

**Comment by busheezy** [comment]
I'm getting the same results.

**Comment by PerfectLaugh** [comment]
I think this is csgo client problem, I reversed the engine code and found nothing special in the OnClientCommandKeyValues

**Comment by PerfectLaugh** [comment]
Nope. I found this is still a KeyValues problem

**Comment by Sikarii** [comment]
Can confirm this is now fixed after your pull request, thanks!

**Comment by Sikarii** [comment]
For anyone looking into this issue, it should be resolved in SM 1.10 build 6545 and 1.11 build 6903.
